felix/bpf/maps/maps.go

```diff
@@ -44,6 +41,49 @@ var (
 	cachedNumPossibleCPUsOnce sync.Once
 )
 
+// Map which map is in which obj file
+var bpfMapObjMap = map[string]string{
+	"cali_counters":         "common_map_stub.o",
+	"cali_counters_scratch": "common_map_stub.o",
+	"cali_iface":            "common_map_stub.o",
+	"cali_jump":             "common_map_stub.o",
+	"cali_perf_evnt":        "common_map_stub.o",
+	"cali_profiling":        "common_map_stub.o",
+	"cali_progs":            "common_map_stub.o",
+	"cali_rule_ctrs":        "common_map_stub.o",
+	"cali_state":            "common_map_stub.o",
+
+	"xdp_cali_jump":  "xdp_map_stub.o",
+	"xdp_cali_progs": "xdp_map_stub.o",
+
+	"cali_v4_arp":     "ipv4_map_stub.o",
+	"cali_v4_ccq":     "ipv4_map_stub.o",
+	"cali_v4_ct":      "ipv4_map_stub.o",
+	"cali_v4_ct_nats": "ipv4_map_stub.o",
+	"cali_v4_frags":   "ipv4_map_stub.o",
+	"cali_v4_frgfwd":  "ipv4_map_stub.o",
+	"cali_v4_frgtmp":  "ipv4_map_stub.o",
+	"cali_v4_fsafes":  "ipv4_map_stub.o",
+	"cali_v4_ip_sets": "ipv4_map_stub.o",
+	"cali_v4_nat_aff": "ipv4_map_stub.o",
+	"cali_v4_nat_be":  "ipv4_map_stub.o",
+	"cali_v4_nat_fe":  "ipv4_map_stub.o",
+	"cali_v4_routes":  "ipv4_map_stub.o",
+	"cali_v4_srmsg":   "ipv4_map_stub.o",
+
+	"cali_v6_arp":     "ipv6_map_stub.o",
+	"cali_v6_ccq":     "ipv6_map_stub.o",
+	"cali_v6_ct":      "ipv6_map_stub.o",
+	"cali_v6_ct_nats": "ipv6_map_stub.o",
+	"cali_v6_fsafes":  "ipv6_map_stub.o",
+	"cali_v6_ip_sets": "ipv6_map_stub.o",
+	"cali_v6_nat_aff": "ipv6_map_stub.o",
+	"cali_v6_nat_be":  "ipv6_map_stub.o",
+	"cali_v6_nat_fe":  "ipv6_map_stub.o",
+	"cali_v6_routes":  "ipv6_map_stub.o",
+	"cali_v6_srmsg":   "ipv6_map_stub.o",
+}
```

Comment:
Almost feels like you can say

if cali_v4* then "ipv4_map_stub.o"
elif cali_v6* then ""ipv6_map_stub.o"
elif xdp* then "xdp_map_stub.o"
else "common_map_stub.o"


The benefit would be that adding a new map would not need to make a change here :thinking:

Reply:
Right, excellent idea! It wouldn't have worked at first because I was relying on this map for maps that are not present in the stubs (via verifying the `ok` var), but now that the logic was reworked to fall back to non-BTF, this will work (and make things simpler)! Will do!